Bugfix: set cli.manualOverride when env var not empty

WeiZhang555 opened the pull request:
<del>In function `NewEnvClient`, the `cli.manualOverride` will be set to true
always which is wrong, to fix it we should let `manualOverride` set to
true only if `os.Getenv("DOCKER_API_VERSION")` isn't empty.

<del>One more step forward, even user set `os.Getenv("DOCKER_API_VERSION")`,
we should still allow updating client API version, in case that user set
a unreasonable version number.

If env var "DOCKER_API_VERSION" is specified by user, we'll set
`cli.manualOverride`, before this, this field is always true due to
wrong logic.

Signed-off-by: Zhang Wei <zhangwei555@huawei.com>

WeiZhang555 commented:
>In function `NewEnvClient`, the `cli.manualOverride` will be set to true
always which is wrong,

You can see it in this piece of code:

```
 	version := os.Getenv("DOCKER_API_VERSION")
 	if version == "" {
 		version = DefaultVersion
 	}
 
 	cli, err := NewClient(host, version, client, nil)
  	if err != nil {
  		return cli, err
  	}
 	if version != "" {
 		cli.manualOverride = true
 	}
```

`version` is never an empty string, which will make `cli.manualOverride = true`

ping related person: @vieux

vieux commented:
@WeiZhang555 it's is not set to true always, it's only set to true when the env var is not specified.

This is the intended behavior, is you specify a version in the env, we should not try to use anything else.

WeiZhang555 commented:
@vieux I mean in this piece of code:
```
 	version := os.Getenv("DOCKER_API_VERSION")
 	if version == "" {
 		version = DefaultVersion
 	}
```

if `os.Getenv("DOCKER_API_VERSION")` is empty, the version will be set to `DefaultVersion` (1.25), after that

```
    if version != "" {
 		cli.manualOverride = true
 	}
```
the if statement will always be true

WeiZhang555 commented:
> This is the intended behavior, is you specify a version in the env, we should not try to use anything else.

I can accept it if this is exactly the design, what I'll suggest is modify this line: 

https://github.com/docker/docker/blob/master/client/client.go#L125-L127

instead:
```
	if os.Getenv("DOCKER_API_VERSION") != "" {
		cli.manualOverride = true
	}
```

WeiZhang555 commented:
> it's is not set to true always, it's only set to true when the env var is not specified.

Ooops, you mean env var **is not** specified?  I thought it should be **is** specified.  Are you making a typo or you really mean it ?

vieux commented:
@WeiZhang555 you're right, it should be 

```
if os.Getenv("DOCKER_API_VERSION") != "" {
		cli.manualOverride = true
	}
```

we set the manualOverride when `DOCKER_API_VERSION` is specified

thaJeztah commented:
Hm, looks like I commented something like that, but it got lost in the last change there? https://github.com/docker/docker/pull/27745#discussion_r87105090

WeiZhang555 commented:
Let do make some update to fix it :smile:

thaJeztah commented:
My previous comment may not be related here (was discussing with @vieux) :blush: Thanks for addressing this @WeiZhang555

WeiZhang555 committed:
Bugfix: set cli.manualOverride when env var not empty

If env var "DOCKER_API_VERSION" is specified by user, we'll set
`cli.manualOverride`, before this, this field is always true due to
wrong logic.

Signed-off-by: Zhang Wei <zhangwei555@huawei.com>

WeiZhang555 commented:
Updated, PTAL~~

vieux commented:
LGTM thanks @WeiZhang555

vdemeester reviewed (approved):
LGTM :cow:
